lib/sync-service.ts

```diff
@@ -0,0 +1,64 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+import { syncEventEmitter } from './sync';
+import {
+  SessionSyncCommand,
+  StorageSyncCommand,
+} from './raft-commands';
+import { Raft, State } from './raft';
+const zluxUtil = require('./util');
+const raftLog = zluxUtil.loggers.raftLogger;
+
+export class SyncService {
+  constructor(
+    private raft: Raft,
+  ) {
+    this.init();
+  }
+
+  private init(): void {
+    const sessionChangeListener = this.onSessionChange.bind(this);
+    const storageChangeListener = this.onStorageChange.bind(this);
+    this.raft.stateEmitter.on('state', (state: State) => {
```

Comment:
During the election Raft ignores all changes, and it should be ok because API Gateway routes traffic only to a leader. I mean, none of instances receive traffic during Leader Election.

Reply:
During the election Raft ignores all changes, and it should be ok because API Gateway routes traffic only to a leader. I mean, none of instances receive traffic during Leader Election.